packages/fluentui/docs/src/examples/components/Card/Usage/CardExampleSelectableGrid.tsx

```diff
@@ -0,0 +1,134 @@
+  return (
+    <Card
+      id={`card${index}`}
+      accessibility={cardSelectableBehavior}
+      aria-labelledby={`card${index} ${selectedMessageId}`}
+      aria-roledescription="user card"
+      onClick={() => {
+        onClick(!selected, index);
+      }}
+      selected={selected}
+      {...unhadledProps}
+    >
+      <Card.Header>
+        <Text content={title} weight="bold" />
+      </Card.Header>
+      <Card.TopControls>
+        <Checkbox
+          accessibility={hiddenComponentBehavior}
+          checked={selected}
+          onClick={(event, props) => {
+            event.preventDefault();
+            onClick(props.checked, index);
+          }}
+        />
+        <div id={selectedMessageId} style={screenReaderContainerStyles} aria-live="polite" role="presentation">
+          {selected ? selectedMessage : notSelectedMessage}
+        </div>
+      </Card.TopControls>
+
+      <Card.Body>
+        <Flex column gap="gap.small">
+          <Image src="public/images/wireframe/square-image.png" />
+        </Flex>
+      </Card.Body>
+    </Card>
+  );
+};
+
+type SelectableCardsGridActions =
+  | { type: 'TOGGLE_ITEM'; selected: boolean; index: string }
+  | { type: 'TOGGLE_ALL'; selected: boolean };
+
+type SelectableCardsGridState = Record<string, boolean>;
+
+const selectableCardsGridStateReducer: React.Reducer<SelectableCardsGridState, SelectableCardsGridActions> = (
+  state,
+  action,
+) => {
+  switch (action.type) {
+    case 'TOGGLE_ITEM': {
+      return { ...state, [action.index]: action.selected };
+    }
+    case 'TOGGLE_ALL': {
+      return _.mapValues(state, () => action.selected);
+    }
+  }
+};
+
+const CardExampleSelectableGrid = () => {
+  // Component setup
+  const cardsNumber = 12;
+  const cards = Array(cardsNumber)
+    .fill(undefined)
+    .map((item, index) => ({ index: index + 1, title: `User ${index + 1}` }));
+  const initialState: SelectableCardsGridState = {
+    ...cards.reduce((cards, card) => {
+      cards[card.index] = false;
+      return cards;
+    }, {}),
+  };
+  const [state, dispatch] = React.useReducer(selectableCardsGridStateReducer, initialState);
+
+  return (
+    <>
+      <Button
+        content="Select all"
+        onClick={() => {
+          dispatch({ type: 'TOGGLE_ALL', selected: true });
+        }}
+      />
+      <Button
+        content="Unselect all"
+        onClick={() => {
+          dispatch({ type: 'TOGGLE_ALL', selected: false });
+        }}
+      />
+      <Grid accessibility={cardsContainerBehavior} columns="3">
+        {cards.map(card => {
+          return (
+            <SelectableCard
+              key={card.index}
+              index={card.index}
+              title={card.title}
+              aria-label={`${card.title} ${card.index} of ${cardsNumber}`}
+              onClick={(isSelected, index) => {
+                dispatch({ type: 'TOGGLE_ITEM', selected: isSelected, index: index });
+              }}
```

Comment:
```suggestion
              onClick={(isSelected, index) => {
                dispatch({ type: 'TOGGLE_ITEM', selected: isSelected, index: card.index });
              }}
```

It looks that we can get `index` from the current scope and `SelectableCard` will not need special signature for `onClick`